record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java

```diff
@@ -82,10 +84,11 @@
         addToStringMethod();
         addHashCodeMethod();
         addEqualsMethod();
-        recordComponents.forEach(component -> {
+        IntStream.range(0, recordComponents.size()).forEach(index -> {
+            var component = recordComponents.get(index);
             add1Field(component);
-            add1SetterMethod(component);
-            add1GetterMethod(component);
+            add1SetterMethod(component, index);
+            add1GetterMethod(component, index);
```

Comment:
There's nothing wrong with a good, old fashioned for-loop if what you're trying to do is go through the indexes of a collection and access both the element and the index.
```java
for (var index = 0; index < recordComponents.size(); index++) {
    var component = recordComponents.get(index);
    add1Field(component);
    add1SetterMethod(component, index);
    add1GetterMethod(component, index);
}
```
Probably much easier to read for most people than using an IntStream.